dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -125,11 +125,16 @@ def authenticate
     # in a new tab. This flow appends a 'new_tab=true' query param, so it will
     # pass this block once the iframe "jail break" has happened.
     if Policies::Lti.force_iframe_launch?(decoded_jwt[:iss]) && !params[:new_tab]
-      id_token = params[:id_token]
-      state = params[:state]
-      token_hash = {id_token: id_token, state: state}
-      redirect_to "#{lti_v1_iframe_url}?#{token_hash.to_query}"
-      return
+      auth_url_base = CDO.studio_url('/lti/v1/authenticate', CDO.default_scheme)
+
+      query_params = {
+        id_token: ERB::Util.url_encode(params[:id_token]),
+        state: ERB::Util.url_encode(params[:state]),
+        new_tab: ERB::Util.url_encode("true"),
```

Comment:
manually URL encoding each parameter seems unnecessary. Is the intent here to do double URL encoding?

wouldn't URL encoding the entire hash be more standard?

```
require 'uri'

# Define the hash
params = {
  name: "John Doe",
  age: 28,
  city: "New York",
  bio: "Software engineer & coffee lover"
}

# Convert the hash to a URL-encoded query string
encoded_params = URI.encode_www_form(params)
```